lib/room.h

```diff
@@ -508,6 +508,19 @@ class Room : public QObject {
     Q_INVOKABLE const Quotient::StateEventBase*
     getCurrentState(const QString& evtType, const QString& stateKey = {}) const;
 
+    /// Get all state events in the room.
+    /*! This method returns all known state events that have occured in
+     * the room, as a mapping from the event type and state key to value.
+     */
+    Q_INVOKABLE const QHash<StateEventKey, const StateEventBase*> stateEvents() const;
```

Comment:
As much as I'd like it to work, I don't think `Q_INVOKABLE` is very useful here: AFAIK `QHash` is not converted to anything on QML/JavaScript side, you'd have to use `QVariantMap` instead. As conversion to `QVariantMap` is quite straightforward I'd prefer to leave that to clients and just give access to the original hash-map.
```suggestion
    const QHash<StateEventKey, const StateEventBase*>& currentState() const;
```